addons/mail/static/src/core/common/store_service.js

```diff
@@ -186,6 +186,9 @@ export class Store extends BaseStore {
         if (ev.target.closest(".o_channel_redirect") && model && id) {
             ev.preventDefault();
             const thread = this.Thread.insert({ model, id });
+            if (!thread.type) {
+                this.env.services["mail.thread"].fetchChannel(id);
+            }
```

Comment:
Would be best to `joinChat` / `joinChannel`, so that the `is_pinned` is properly set

Reply:
I thought the intention was to prevent members from joining with just a click and they should only join when performing a specific action or manually opting in

Reply:
There's no such thing except with the welcome page of the public discuss page.
Opening a conversation and seeing messages must require joining the conversation with current specs.

Fetching channel data and putting `bus_service.addChannel(thread.busChannel)` is a hacky way to do simulate a "join", albeit without "X has joined" notification and chat window is lost on page reload / doesn't sync among tabs due to lack of `is_pinned: true`.

We want the "X has joined" notif and tabs syncing / chat window kept on page reload, so the "join" is preferred over this hack.

Reply:
But it makes it that you would end up joining channels even if you didn't intend to. You would join channel when clicking on it's mention.

Reply:
I didn't post as a reply of this discussion but just to repeat it at the correct place I don't think we want to join in this version. We can improve the spec in master if necessary. So the current fix would be ok, just missing `await` on fetch.